specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2024-05-01-preview/appplatform.json

```diff
@@ -9738,6 +9999,155 @@
         }
       }
     },
+    "BuildSettingResourceCollection": {
+      "description": "Object that includes an array of build setting resources and a possible link for next set",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of build setting resources",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BuildSettingResource"
+          }
+        },
+        "nextLink": {
+          "description": "URL client should use to fetch the next page (per server side paging).\r\nIt's null for now, added for future use.",
+          "type": "string"
+        }
+      }
+    },
+    "BuildSettingResource": {
+      "description": "Build setting resource object",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/BuildSettingProperties",
+          "description": "Properties of a build setting",
+          "x-ms-client-flatten": false
+        }
+      }
+    },
+    "BuildSettingProperties": {
+      "description": "Properties of a build setting",
+      "type": "object",
+      "required": [
+        "type"
+      ],
+      "properties": {
+        "type": {
+          "description": "build setting type",
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "State of the build setting.",
+          "enum": [
+            "Creating",
+            "Updating",
+            "Succeeded",
+            "Failed",
+            "Deleting",
+            "Canceled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "BuildSettingProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "secrets": {
```

Comment:
control plane hard limits payload size  to 4MB. Will the 3 files exceed the limitation?